src/templates/shared/navbar.html

```diff
@@ -23,9 +23,26 @@
     <div class="collapse navbar-collapse" collapse="collapsed">
       <!-- left -->
       <div class="navbar-left">
-        <form class="navbar-form" role="search">
+        <form
+          name="searchForm"
+          class="navbar-form"
+          role="search"
+          ng-submit="searchForm.$valid && phSearch(search.body, 10)"
+          novalidate
+          >
           <div class="input-group">
-            <input type="text" class="form-control" placeholder="Search PaperHub" name="srch-term" id="srch-term">
+            <input
+              type="text"
+              class="form-control"
+              placeholder="Search PaperHub"
+              name="srch-term"
+              id="srch-term"
+              ng-model="search.body"
+              typeahead="result.title for result in phSearch($viewValue, 5)"
+              typeahead-wait-ms="250"
+              typeahead-on-select="goToArticle($item, $model, $label)"
+              required
+              >
             <div class="input-group-btn">
```

Comment:
maybe we should remove the button and place the icon in the input field (like [here](http://getbootstrap.com/css/#forms-control-validation))